backend/wayland/seat.c

```diff
@@ -687,6 +719,9 @@ void create_wl_pointer(struct wlr_wl_seat *seat, struct wlr_wl_output *output) {
 		pointer->gesture_pinch = zwp_pointer_gestures_v1_get_pinch_gesture(
 				backend->zwp_pointer_gestures_v1, wl_pointer);
 		zwp_pointer_gesture_pinch_v1_add_listener(pointer->gesture_pinch, &gesture_pinch_impl, dev);
+		pointer->gesture_hold = zwp_pointer_gestures_v1_get_hold_gesture(
```

Comment:
We need to check the version supported by the compositor here, and avoid sending this request if it's < 3. You can get the version with `zwp_pointer_gestures_v1_get_version`.

We also need to request v3 if supported by the compositor in `registry_global`.

We're also missing the cleanup logic, but all the other gesture types are missing it too, so arguably that's unrelated to this PR.

Reply:
> We're also missing the cleanup logic, but all the other gesture types are missing it too, so arguably that's unrelated to this PR.

https://github.com/swaywm/wlroots/pull/3048